memgpt/llm_api/openai.py

```diff
@@ -263,7 +287,10 @@ def _sse_post(url: str, data: dict, headers: dict) -> Generator[ChatCompletionCh
                     response_dict = json.loads(response_bytes.decode("utf-8"))
                     error_message = response_dict["error"]["message"]
                     # e.g.: This model's maximum context length is 8192 tokens. However, your messages resulted in 8198 tokens (7450 in the messages, 748 in the functions). Please reduce the length of the messages or functions.
-                    raise Exception(error_message)
+                    if "maximum context length" in error_message:
```

Comment:
@sarahwooders we should probably move this to `constants.py`? It's currently how we identify a max context error.

For non-OpenAI backends (eg LMStudio), I manually checked what the context window overflow message was and wrote a handler (inside of the "local LLM" code) which catches the different message, and re-raises it with `maximum context length` in the error message instead so that it gets caught in `agent.py`.